projects/dyson-money/index.js

```diff
@@ -61,30 +36,43 @@ async function polygonTvl(timestamp, block, chainBlocks) {
   return balances;
 } 
 
-async function arbitrumTvl(timestamp, block, chainBlocks) {
-  let balances = {};
-  await unwrapUniswapV3NFTs({balances, owners: arbitrumPools, block: chainBlocks.arbitrum, chain: 'arbitrum'})
-  return balances;
-}
+let _response
 
-async function optimismTvl(timestamp, block, chainBlocks) {
-  let balances = {};
-  await unwrapUniswapV3NFTs({balances, owners: optimismPools, block: chainBlocks.optimism, chain: 'optimism'})
-  return balances;
+function fetchChain(chainId) {
+  return async () => {
+    if (!_response) _response = utils.fetchURL('https://api.dyson.money/tvl')
+    const response = await _response;
+
+    let tvl = 0;
+    const chain = response.data[chainId];
+    for (const vault in chain) {
+      tvl += Number(chain[vault]);
+    }
+    if (tvl === 0) {
+      throw new Error(`chain ${chainId} tvl is 0`)
+    }
+
+    return toUSDTBalances(tvl);
+  }
 }
 
+const chains = {
+  optimism: 10,
+  polygon: 137,
+  arbitrum: 42161,
+}
 
 module.exports = {
   misrepresentedTokens: false,
   methodology: "TVL is calculated by summing the liquidity in the Uniswap V3 pools.",
   polygon: {
-    tvl: polygonTvl,
+    tvl: polygonTvl && fetchChain(137),
```

Comment:
I think this would only take polygonTvl or ` fetchChain(137)`  value, not both

Reply:
@g1nt0ki Yeh, it only takes one of the values (in this case the one with a higher TVL). Will submit a new PR soon so that everything falls under the one function

Reply:
(Tired me somehow didn’t pick that up)